tests/test_typeinfo.py

```diff
@@ -44,29 +48,75 @@ async def test_fetch_async(aconn, name, status):
     assert info.array_oid == psycopg.adapters.types["text"].array_oid
 
 
-@pytest.mark.parametrize("name", ["nosuch", sql.Identifier("nosuch")])
-@pytest.mark.parametrize("status", ["IDLE", "INTRANS"])
-def test_fetch_not_found(conn, name, status):
+_name = pytest.mark.parametrize("name", ["nosuch", sql.Identifier("nosuch")])
+_status = pytest.mark.parametrize("status", ["IDLE", "INTRANS"])
+_info_cls = pytest.mark.parametrize(
+    "info_cls, has_to_regtype",
+    [
+        pytest.param(TypeInfo, True, marks=pytest.mark.crdb_skip("to_regtype")),
+        pytest.param(TypeInfo, False),
+        pytest.param(RangeInfo, True, marks=pytest.mark.crdb_skip("range")),
+        pytest.param(
+            MultirangeInfo,
+            True,
+            marks=(pytest.mark.crdb_skip("range"), pytest.mark.pg(">= 14")),
+        ),
+        pytest.param(CompositeInfo, True, marks=pytest.mark.crdb_skip("composite")),
+        pytest.param(EnumInfo, True, marks=pytest.mark.crdb_skip("to_regtype")),
+        pytest.param(EnumInfo, False),
+    ],
+)
+
+
+@_name
+@_status
+@_info_cls
+def test_fetch_not_found(conn, name, status, info_cls, has_to_regtype, monkeypatch):
+    exit_orig = psycopg.Transaction.__exit__
+
+    if has_to_regtype:
+
+        def exit(self, exc_type, exc_val, exc_tb):
+            assert exc_val is None
+            return exit_orig(self, exc_type, exc_val, exc_tb)
+
+        monkeypatch.setattr(psycopg.Transaction, "__exit__", exit)
+    else:
+        monkeypatch.setattr(TypeInfo, "_has_to_regtype_function", lambda _: False)
```

Comment:
I don't understand the need of this branch. Could you please clarify?

Reply:
the original idea was to test the case of no support for `to_regtype` even for server that do support it.
I guess it could be removed is an old version of cockroach db is always tested (at least in the pipeline)

Reply:
Yes, my plan is for CI to run on all the databases/versions supported, so I don't think that testing the monkeypatched version of the adapter is useful. You might even want to drop the `has_to_regtype` param and use `TypeInfo._has_to_regtype_function()` to decide whether to test `__exit__` or not.

Reply:
ok, let's do that